src/assignments/assignments.controller.ts

```diff
@@ -0,0 +1,27 @@
+import { Controller, Post, UseInterceptors, UploadedFile, Param, Body } from "@nestjs/common";
+import { FileInterceptor } from "@nestjs/platform-express";
+
+import { AssignmentsService } from "./assignments.service";
+
+@Controller("classrooms")
+export class AssignmentsController {
+  constructor(private readonly assignmentsService: AssignmentsService) {}
+
+  @Post(":classroomId/assignments")
+  @UseInterceptors(FileInterceptor("file"))
+  uploadAssignment(
+    @UploadedFile() file: Express.Multer.File,
+    @Param("classroomId") classroomId: number,
+    @Body("title") title: string,
+    @Body("description") description: string,
+    @Body("dueDate") dueDate: string,
```

Comment:
Can't we create a dto class matching the body structure and use that rather than accessing each individually?